### PR TITLE
Github Action to enforce allowed hosts are set for GA and Beta connectors

### LINKED ISSUE #21184: Test that Beta and GA connectors have `allowed_hosts`
Using our `ci_connector_ops` tools, we can enforce that Beta and Alpha connectors which do yet `allowed_hosts` in their actor definitions will fail a CI check.

### PR DESCRIPTION
**Pull request by evantahler**
This PR adds a check to the renamed `Connector Ops CI - Connector Metadata Checks` to ensure that GA connectors include `allowed_hosts` in their actor definition. 

Requires https://github.com/airbytehq/airbyte/pull/21676 to be merged first
Closes #21184

### CODE REVIEW

File: tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        connector_release_stage = utils.get_connector_release_stage(connector_name)
+        if connector_release_stage in RELEASE_STAGES_TO_CHECK:
+          missing = not connector_has_allowed_hosts(connector_name)
+          if missing:
+            connectors_missing_allowed_hosts.append(connector_name)
+
+    return connectors_missing_allowed_hosts
+
+def connector_has_allowed_hosts(connector_name: str) -> bool:
+  definition = utils.get_connector_definition(connector_name)
```
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py:31]: @alafanechere these utilities are great!

File: .github/workflows/connector_metadata_checks.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
-name: Connector Ops CI - Check SAT strictness level
+name: Connector Ops CI - Connector Metadata Checks
```
Comment (evantahler) [inline .github/workflows/connector_metadata_checks.yml:1]: Renaming and extending our existing Github Action

File: tools/ci_connector_ops/setup.py [hunk 3]
```diff
@@ -21,7 +21,8 @@
         "console_scripts": [
```
Comment (alafanechere) [inline tools/ci_connector_ops/setup.py:21]: Would you mind bumping the package version? It's not released anywhere so versionning it a good practice only at the moment.

File: tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  # print("----- " + connector_name  + " -----")
+  # print(definition)
```
Comment (alafanechere) [inline tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py:29]: You can remove these debugs comments, don't you?

File: tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  try:
+    return definition["allowedHosts"] != None
+  except KeyError:
+    return False
```
Comment (alafanechere) [inline tools/ci_connector_ops/ci_connector_ops/allowed_hosts_checks.py:33]: More pythonic
```suggestion
    return definition.get("allowedHosts") is not None
```

**Review by alafanechere — APPROVED** [review]
Nice! I only have a couple of minor suggestions.
I think we might want to eventually move this to a QA check as part of `/test`. This module is ready to welcome it if you like:
https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/qa_checks.py#L71

But as it is release stage specific check I'm fine keeping it here 👍

**Review by erohmensing — APPROVED** [review]

**Pull request merged**